src/main/java/org/gridsuite/filter/server/DanglingLineFilterRepositoryProxy.java

```diff
@@ -38,17 +44,32 @@ public DanglingLineFilterRepository getRepository() {
 
     @Override
     public AbstractFilter toDto(DanglingLineFilterEntity entity) {
-        return buildInjectionFilter(
-            DanglingLineFilter.builder(), entity).build();
+        return new FormFilter(
+            entity.getId(),
+            entity.getCreationDate(),
+            entity.getModificationDate(),
+            new DanglingLineFilter(
+                entity.getEquipmentId(),
+                entity.getEquipmentName(),
+                entity.getSubstationName(),
+                setToSorterSet(entity.getCountries()),
+                convert(entity.getNominalVoltage())
+            )
+        );
     }
 
     @Override
     public DanglingLineFilterEntity fromDto(AbstractFilter dto) {
-        if (dto instanceof DanglingLineFilter) {
-            var danglingLineFilterEntityBuilder = DanglingLineFilterEntity.builder();
-            buildInjectionFilter(danglingLineFilterEntityBuilder, (DanglingLineFilter) dto);
-            return danglingLineFilterEntityBuilder.build();
+        if (!(dto instanceof FormFilter)) {
+            throw new PowsyblException(WRONG_FILTER_TYPE);
+        }
```

Comment:
I see many of these line shouldn't 
FormFilter toFormFilter(AbstractFilter abstractFilter) { 
if (!(dto instanceof FormFilter)) {
            throw new PowsyblException(WRONG_FILTER_TYPE);
        }
        return (FormFilter) dto; 
}
do the job ?

Reply:
Done